### DISCUSSION TITLE
Number of connections

### DISCUSSION THREAD

**Discussion by btazus** [iterate-ch/cyberduck#14374] [Ideas]
In the CLI, there is an option for --parallel where I can choose the number of connections I can make.
I would really hope to have the same for Mountain Duck.

I use parallel connections with the Google Workspace API on other projects and it works well.
I, as admin, did have to ask Google to increase my API quota, but they did so readily.

**Comment by AliveDevil** [comment]
Mountain Duck uses as many connections as required to fulfil filesystem requests as fast as possible.
e.g. in Online mode a file explorer copy might be sequential, thus no parallel connections are established.
On Windows Mountain Duck would use multiple connections when multiple files are written in parallel - e.g. `robocopy /MT:4` would copy 4 files in parallel.
In Smart Sync mode this can be optimized to upload all files after the copy operation finishes.

**Reply by dkocher** [reply]
>  In Smart Sync mode this can be optimized to upload all files after the copy operation finishes.

The actual number of parallel connections used in Mountain Duck for file transfers in the background in _Smart Synchronization_ connect mode is set to `10` by default. However this only comes into play when there are a large number of files queued that can be coalesced together into a single transfer.
